README.md

```diff
@@ -1,168 +1,222 @@
+On MacOs, install node with [Homebrew](https://brew.sh).
+For other platforms, please visit the [Node.js download page](https://nodejs.org/en/download/).
 
 ```bash
-$ node -v
-$ npm -v
+brew install node@12
 ```
 
-Install [Node](https://nodejs.org/en/download/) that includes npm in version 10.9.0 or later:
+_Developer hint: To switch between various node versions, we recommand to use [n &mdash; Node.js versions manager](https://www.npmjs.com/package/n)._
+
+To install it:
 
 ```bash
-$ brew install node
+npm install -g n
 ```
+and switch to node version, e.g. 12.16.2 with `n v12.16.2`
 
-For other platforms, please go to the Node.js website.
+#### NPM package manager
 
-The easiest way to install node
-in the correct version is to use ['n'](https://github.com/tj/n):
+We use the [npm](https://docs.npmjs.com/cli/install) instead of yarn, which is installed with Node.js by default.
 
-```bash
-$ npm install -g n
-$ n v10.9.0
-```
+To check that you have the npm client installed, run `npm -v`.
 
 ### First steps
 
 Install the node packages with:
 
 ```bash
-$ npm install
+npm install
 ```
 
 and build the library with:
 
 ```bash
-$ npm run build-lib
+npm run build-lib
 ```
 
-### Develop
+### Contribution
 
 If you want to add more components, services and so on to a module of the library, you can do it with:
 
-`$ ng generate component [path/in/your/module/][name-of-component] --project @knora/ui --styleext scss --prefix kui`
+```bash
+ng generate component [path/in/the/module/][name-of-component] --project @dasch-swiss/dsp-ui
+```
 
-Before testing the new component inside of the demo app, you have to rebuild after each change: `ng build @knora/ui`.
+For example:
 
-#### Run the DSP-UI Demo Application
+```bash
+ng generate component core/test --project @dasch-swiss/dsp-ui
+```
 
-Run the app with `ng s`. The demo app runs on <http://0.0.0.0:4200> and we use it for documentation on [Knora-ui Github page](https://dasch-swiss.github.io/knora-ui).
+will create a component-folder called `test` inside of `projects/dsp-ui/src/lib/core/` with four files:
 
-There's a test environment for the modules on <https://github.com/dhlab-basel/knora-ui-playground> with yalc.
+- `test.component.scss`
+- `test.component.html` 
+- `test.component.spec.ts`
+- `test.component.ts`
 
-* * *
+The main component file should look as follow:
 
-#### YALC
+```typescript
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'kui-test',
+  templateUrl: './test.component.html',
+  styleUrls: ['./test.component.scss']
+})
+export class TestComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+}
+```
 
-> Better workflow than `npm` \| `yarn` link for package authors.
+Before testing the new component inside of the demo app, you have to rebuild after each change: 
 
-Yalc publishes the packages to a local store (not the npm website).
+```bash
+npm run build-lib
+```
+
+### Run the DSP-UI-LIB Demo Application
```

Comment:
Could you move this section up (before contribution)?

Reply:
Why? This is part of contribution

Reply:
if mean put it after "running app in dev mode", sorry if this was unlcear